### PR TITLE
Tests now cover all the installed_product.go file

### PR DESCRIPTION
**Pull request by mssola**

### CODE REVIEW

**Comment by mssola** [comment]
The `baseProductLoc` was a bit of a problem if we wanted to seriously test this file. Therefore, instead we're doing it through an interface, so it can then be mocked in tests. As you can see, I create a simple mock for each different case that we might encounter. The big xml string inside the previous test has been moved to a proper file.

Moreover, I've also added some documentation.

**Comment by mssola** [comment]
Of course, tests are still failing because with this change we just rose from 40% to 50% code coverage, so it's still not 100%

File: installed_product.go [hunk 1]
```diff
@@ -36,33 +49,40 @@ func (p InstalledProduct) String() string {
...
-
 	return p, nil
 }
 
-// read the product file from the standard location
-func ReadInstalledProduct() (InstalledProduct, error) {
-	if _, err := os.Stat(baseProductLoc); os.IsNotExist(err) {
+// Read the product file from the standard location
+func readInstalledProduct(b ProductProvider) (InstalledProduct, error) {
+	if _, err := os.Stat(b.Location()); os.IsNotExist(err) {
```
Comment (flavio) [inline installed_product.go]: I don't like the variable being named `b`, it's not helpful.

**Comment by flavio** [comment]
:+1:

File: installed_product.go (hunk 1 above)
Comment (mssola) [inline installed_product.go]: @flavio that's the result of copy-pasting :P I'll fix that tomorrow

**Pull request merged**